### PR TITLE
Fix Mac standalone window focus

### PR DESCRIPTION
**Pull request by lucianoiam**
CI build is failing but seems unrelated to this patch

```
-- Checking for module 'liblo'
38
--   No package 'liblo' found
39
-- Configuring incomplete, errors occurred!
40
See also "/Users/runner/work/DPF/build/CMakeFiles/CMakeOutput.log".
41
Error: Process completed with exit code 1.
```

### CODE REVIEW

File: dgl/src/ApplicationPrivateData_macOS.mm [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#import <AppKit/AppKit.h>
+
+#include "ApplicationPrivateData.hpp"
+
+START_NAMESPACE_DGL
+
+void macActivateApp()
+{
+    [NSApp setActivationPolicy:NSApplicationActivationPolicyRegular];
+    [NSApp activateIgnoringOtherApps:YES]; // focus window
```
Comment (falkTX) [inline dgl/src/ApplicationPrivateData_macOS.mm:26]: this is called on other places in pugl, no? I dont think this line in specific is needed.

**Comment by falkTX** [comment]
Thanks for the PR, but the setup with these files is not good.
Please put the new function in pugl.hpp/cpp instead, there are already a couple of OS-specific functions there.

**Comment by falkTX** [comment]
Actually maybe this needs to be in pugl directly?
Is there a reason not to?

**Comment by lucianoiam** [comment]
Thanks for reviewing. if the standalone plugins were built as app bundles (akin to the VST2 bundle) this call wouldn't be necessary. The ultimate fix is providing a .app bundle. I don't think Pugl should assume how the binaries are to be distributed (bare executable vs. bundle). This is more like a stopgap measure. 

Would it be acceptable to move `macActivateApp()` to pugl.mm, get rid of the new .mm file and submit a new PR ?

**Comment by falkTX** [comment]
> Would it be acceptable to move `macActivateApp()` to pugl.mm, get rid of the new .mm file and submit a new PR ?

Yes, just name it in the same style as the existing extra functions stuff.

I think the issue should be brought to pugl too in any case.
That call doesn't hurt if running through an app bundle, yes?

**Comment by lucianoiam** [comment]
If this is moved to Pugl, both NSApp methods should be called once during application startup. Pugl currently calls `activateIgnoringOtherApps` in `puglRealize()`, if I correctly understand that is each time a new window is created, not sure about the rationale behind that.

Just tested and patch does not hurt when running as bundle, here is a quick test showing that and also how wrapping binary in .app fixes the issue without needing the NSApp calls.

[d_states.zip](https://github.com/DISTRHO/DPF/files/6940493/d_states.zip)

**Comment by lucianoiam** [comment]
Also note Pugl does not call `[NSApp setActivationPolicy:NSApplicationActivationPolicyRegular]` which is also needed.

**Pull request closed without merging**